src/navigation/navigation-events.js

```diff
@@ -94,6 +94,12 @@ function urlReroute() {
 function patchedUpdateState(updateState, methodName) {
   return function () {
     const urlBefore = window.location.href;
+    
+    // issues/1030 - Vue requires 'current' to be set at all times
+    if (arguments && arguments[0] && !arguments[0]?.current) {
```

Comment:
I prefer not using the `arguments` keyword. Also, the optional chaining on this line seems unnecessary since there's already a check for whether `arguments[0]` is truthy.

I'd prefer something like this:

```js
return function (state) {
  if (state && !state.current) {
    state.current = "";
  }  
}
```

Reply:
Hi @joeldenning ,
i tried solving this on the vue-router side but there seemed to be no flexibility on their end to fix it https://github.com/vuejs/router/pull/1502

Regarding the arguments, where do i get the "state" object?
The arguments seem to be directly applied to the **_window.history.pushState_** or **_window.history.replaceState_** on the code:
`const result = updateState.apply(this, arguments);
`